types/whatwg-url/index.d.ts

```diff
@@ -1,90 +1,106 @@
-// Type definitions for whatwg-url 6.4
+// Type definitions for whatwg-url 8.0
 // Project: https://github.com/jsdom/whatwg-url#readme
 // Definitions by: Alexander Marks <https://github.com/aomarks>
+//                 ExE Boss <https://github.com/ExE-Boss>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
 
 /// <reference types="node"/>
 
 /** https://url.spec.whatwg.org/#url-representation */
 export interface URLRecord {
-    scheme: string;
-    username: string;
-    password: string;
-    host: string|number|IPv6Address|null;
-    port: number|null;
-    path: string[];
-    query: string|null;
-    fragment: string|null;
-    cannotBeABaseURL?: boolean;
+	scheme: string;
+	username: string;
+	password: string;
+	host: string | number | IPv6Address | null;
+	port: number | null;
+	path: string[];
+	query: string | null;
+	fragment: string | null;
+	cannotBeABaseURL?: boolean;
 }
 
 /** https://url.spec.whatwg.org/#concept-ipv6 */
-export type IPv6Address =
-    [number, number, number, number, number, number, number, number];
+export type IPv6Address = [number, number, number, number, number, number, number, number];
 
 /** https://url.spec.whatwg.org/#url-class */
 export class URL {
-    constructor(url: string, base?: string);
-
-    href: string;
-    readonly origin: string;
-    protocol: string;
-    username: string;
-    password: string;
-    host: string;
-    hostname: string;
-    port: string;
-    pathname: string;
-    search: string;
-    readonly searchParams: URLSearchParams;
-    hash: string;
-
-    toJSON(): string;
+	constructor(url: string, base?: string | URL);
```

Comment:
Taking a URL object for `base` here does not appear to be in the spec (see https://url.spec.whatwg.org/#url-class which says it's a `USVString`), though I confirmed it does actually work both in this Node library and in Chrome.

Reply:
That’s&nbsp;because the&nbsp;URL gets&nbsp;`string`ified.